src/database/database-internal.ts

```diff
@@ -28,9 +28,13 @@ import { getSdkVersion } from '../utils/index';
 
 import Database = database.Database;
 
+const TOKEN_REFRESH_THRESHOLD_MILLIS = 5 * 60 * 1000;
+
 export class DatabaseService {
 
   private readonly appInternal: FirebaseApp;
+  private tokenRefresh: boolean;
```

Comment:
This doesn't sound like a boolean variable to me (but naming is hard and criticizing is easy)

Reply:
`refreshToken` or `shouldRefreshToken` might work... I agree with "naming is hard and criticizing is easy" :)

Reply:
Renamed to `tokenListenerRegistered`. See below.